client/data/users/with-users.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import useUsersQuery from './use-users-query';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const withUsers = createHigherOrderComponent(
+	( Wrapped, fetchOptions = {}, queryOptions = {} ) => ( props ) => {
+		const siteId = useSelector( getSelectedSiteId );
+		const usersQuery = useUsersQuery( siteId, fetchOptions, queryOptions );
```

Comment:
`useUsersQuery` is using `useInfiniteQuery` under the hood which provides pagination handlers. Do we ever need to call [`fetchNextPage`](https://react-query.tanstack.com/reference/useInfiniteQuery#_top)? I'm not exactly sure about previous behaviour.

Reply:
Not for this use case, but we might need it in the future.